apps/extension/src/entrypoints/background.ts

```
@@ -407,8 +417,15 @@ async function handle(request: Request): Promise<ResponseMap[Request['type']]> {
       // Rightmove only. The URL comes from a content script, and a content script is running in a
       // page whose scripts we do not control; a worker that opened whatever it was handed would
       // be a redirector for anything that got a message into it.
-      if (!/^https:\/\/www\.rightmove\.co\.uk\/properties\/\d+/.test(request.url)) {
-        throw new Error(`refusing to open ${request.url} — only Rightmove listings`);
+      //
+      // Two shapes and no more: a listing, and a rental search (`find.html`, the one page the sweep
+      // panel runs on). The second is what lets the website's unattended sweep page through a
+      // neighbourhood's results the way a person would — each page is a real navigation in a real
+      // background tab, recorded by the same panel that records a page you opened yourself. Nothing
+      // here fetches a search; see the standing rule in AGENTS.md.
+      const isSearch = SEARCH_URL.test(request.url);
```

Biome (2.5.7)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)